lib/msal-react/src/hooks/useAccount.ts

```diff
@@ -4,42 +4,28 @@
  */
 
 import { useState, useEffect } from "react";
-import { AccountInfo, IPublicClientApplication, InteractionStatus, AccountEntity } from "@azure/msal-browser";
+import { AccountInfo, IPublicClientApplication, AccountEntity } from "@azure/msal-browser";
 import { useMsal } from "./useMsal";
 import { AccountIdentifiers } from "../types/AccountIdentifiers";
+import { getAccountByIdentifiers } from "../utils/utilities";
 
-function getAccount(instance: IPublicClientApplication, accountIdentifiers: AccountIdentifiers): AccountInfo | null {
-    const allAccounts = instance.getAllAccounts();
-    if (allAccounts.length > 0 && (accountIdentifiers.homeAccountId || accountIdentifiers.localAccountId || accountIdentifiers.username)) {
-        const matchedAccounts = allAccounts.filter(accountObj => {
-            if (accountIdentifiers.username && accountIdentifiers.username.toLowerCase() !== accountObj.username.toLowerCase()) {
-                return false;
-            }
-            if (accountIdentifiers.homeAccountId && accountIdentifiers.homeAccountId.toLowerCase() !== accountObj.homeAccountId.toLowerCase()) {
-                return false;
-            }
-            if (accountIdentifiers.localAccountId && accountIdentifiers.localAccountId.toLowerCase() !== accountObj.localAccountId.toLowerCase()) {
-                return false;
-            }
-
-            return true;
-        });
-
-        return matchedAccounts[0] || null;
-    } else {
-        return null;
+function getAccount(instance: IPublicClientApplication, accountIdentifiers?: AccountIdentifiers): AccountInfo | null {
+    if (!accountIdentifiers || (!accountIdentifiers.homeAccountId && !accountIdentifiers.localAccountId && !accountIdentifiers.username)) {
+        // If no account identifiers are provided, return active account
+        return instance.getActiveAccount();
     }
+
+    return getAccountByIdentifiers(instance.getAllAccounts(), accountIdentifiers);
 }
 
 /**
  * Given 1 or more accountIdentifiers, returns the Account object if the user is signed-in
  * @param accountIdentifiers 
  */
-export function useAccount(accountIdentifiers: AccountIdentifiers): AccountInfo | null {
+export function useAccount(accountIdentifiers?: AccountIdentifiers): AccountInfo | null {
```

Comment:
Is there a reason behind the mixed use of null or undefined in this library? As CommonAuthorizationUrlRequest takes `AccountInfo | undefined` it would be prettier if functions returning accountInfo, like this and instance.getActiveAccount() returned undefined.

Reply:
`null` = explicitly none
`undefined` = optional/not provided

Using `null` helps us avoid mistakes when we expect certain variables to be assigned (either to a value or an acknowledgement that it doesnt have one) whereas `undefined` also carries with it the additional possibility that it hasn't been assigned yet, and that may be allowed in some scenarios i.e. optional parameters or it could be a mistake.

To give a more concrete example, in the case of `useAccount` and `getActiveAccount` by using `null` instead of `undefined` we are saying "we checked and we have no account to return to you" vs. "maybe we checked and there was no account or maybe we didnt check at all"